hybrasyl/Board.cs

```diff
@@ -0,0 +1,319 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Data.SqlTypes;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using Microsoft.Scripting.Interpreter;
+using Newtonsoft.Json;
+
+namespace Hybrasyl
+{
+     
+    public class MessageStoreLocked : Exception
+    {
+
+    }
+
+    [JsonObject(MemberSerialization.OptIn)]
+    public class MessageStore : IEnumerable<Message>
+    {
+        [JsonProperty] public string Name;
+        [JsonProperty] public string DisplayName;
+        [JsonProperty] public List<Message> Messages;
+        [JsonProperty] public string Guid;
+        [JsonProperty] public short CurrentId;
+        public int Id;
+
+        public bool Full => Messages.Count == short.MaxValue;
+        private int _lock; 
+        public bool IsSaving;
```

Comment:
I suppose they do not need to be explicit, however, having a private member
variable after the constructor is initialized allows for better internal
class control, at least in my opinion. Either way works just fine.

On Wed, Feb 10, 2016 at 3:15 PM Justin Baugh <notifications@github.com>
wrote:

> In hybrasyl/Board.cs
> <https://github.com/hybrasyl/server/pull/30#discussion_r52518173>:
>
> > +
> > +    }
> > +
> > +    [JsonObject(MemberSerialization.OptIn)]
> > +    public class MessageStore : IEnumerable<Message>
> > +    {
> > +        [JsonProperty] public string Name;
> > +        [JsonProperty] public string DisplayName;
> > +        [JsonProperty] public List<Message> Messages;
> > +        [JsonProperty] public string Guid;
> > +        [JsonProperty] public short CurrentId;
> > +        public int Id;
> > +
> > +        public bool Full => Messages.Count == short.MaxValue;
> > +        private int _lock;
> > +        public bool IsSaving;
>
> Why does this need to be explicit? I thought these were auto properties,
> and the compiler basically generates that code from the simpler declaration
> (https://msdn.microsoft.com/en-us/library/bb384054.aspx)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/hybrasyl/server/pull/30/files#r52518173>.
>